src/account_manager.rs

```diff
@@ -432,15 +432,18 @@ impl AccountManager {
             .map(|i| {
                 Ok(DeviceInfo {
                     id: i.id,
-                    name: i
-                        .name
-                        .map(|s| {
-                            decrypt_device_name_from_device_info(
-                                &s,
-                                &aci_identity_keypair,
-                            )
-                        })
-                        .transpose()?,
+                    name: i.name.map(|s| {
+                        match decrypt_device_name_from_device_info(
+                            &s,
+                            &aci_identity_keypair,
+                        ) {
+                            Ok(name) => name,
+                            Err(e) => {
+                                tracing::error!("{e}");
+                                String::from("N/A")
+                            },
+                        }
+                    }),
```

Comment:
Hmm, what about this?
```suggestion
                    name: i.name.and_then(|s| {
                        match decrypt_device_name_from_device_info(
                            &s,
                            &aci_identity_keypair,
                        ) {
                            Ok(name) => Some(name),
                            Err(e) => {
                                tracing::error!("{e}");
                                None
                            },
                        }
                    }),
```

I think that captures your intention, and doesn't allocate an English-language string, and allows us to handle the device-name-missing without relying on that English-language-string

Reply:
That makes a lot more sense, and not using a magic string is the correct thing to do indeed. I somehow missed the obvious `None` possibility here.